Add `balenalib/raspberrypi3-debian-python:3.11-bookworm` to rpi builds for newer  RPi OS

misl6 opened the pull request:
<!--
Thank you for pull request.

Below are items maintainers should consider when merging the PR. Feel free to suggest a `unit@` label or check-mark the others as appropriate.

-->
Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

The newer Raspberry Pi OS has been released 2 weeks ago, and a new RPi wheel has been already requested via #8416 .

This PR adds `balenalib/raspberrypi3-debian-python:3.11-bookworm` to our build matrix, so a compatible wheel can be produced.

The docs have been updated accordingly, and a note section has been added for headless environments, where some additional dependencies have to be installed.

(When a new Kivy version gets released, we will also need to send this artifact to piwheels maintainer)

Tested on:
✅  Raspberry Pi 4 + Raspberry Pi OS 5.0 32 bit (Bookworm) - Python 3.11
✅ Raspberry Pi 4 + Raspberry Pi OS 5.0 Lite 32 bit (Bookworm) - Python 3.11

p0lygun commented:
@misl6, i think its fine to link this PR to the issue. 

Now you can link an issue to PR by using a closing keyword in pr description (ie `fixes #8416`) or using the tab on right

misl6 commented:
> @misl6, i think its fine to link this PR to the issue.
> 
> Now you can link an issue to PR by using a closing keyword in pr description (ie `fixes #8416`) or using the tab on the right

Linked issue, so it will be closed automatically.

FYI: Moved to WIP, as it segfaults while performing on-device tests.

misl6 commented:
Removed [WIP], the new `patchelf` seems to have introduced a bug, that was leading to a segfault during the loading of a patched library.

Reverted to the previous `patchelf`, and now everything looks good.
